File: EatHealthy/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace EatHealthy.Models
{
    public class Post
    {
        public int ID { get; set; }

        [Required]
        [ForeignKey("Client")]
        [DisplayName("Posting Client")]
        public int ClientID { get; set; }

        [Required]
        [ForeignKey("Category")]
        [DisplayName("Category of the Post")]
        public int CategoryID { get; set; }

        [MaxLength(20)]
        [Required]
        [DisplayName("Title")]
        public string Title { get; set; }

        [MaxLength(5000)]
        [Required]
        public string Content { get; set; }

        [DataType(DataType.Date)]
        [DisplayName("Created at")]
        public DateTime CreationDate { get; set; }

        public virtual Client Client { get; set; }

        public virtual Category Category { get; set; }

        public virtual List<Comment> Comments { get; set; }
    }
}

File: NewEatHealthy/server/models/Post.js
var mongoose = require('mongoose')
  , Schema = mongoose.Schema
  , ObjectId = Schema.ObjectId

  var PostSchema = new Schema({
    title : {
        type :  String, required: true
    },
    content: {
        type :  String, required: true
    },
    creationDate : {
        type : Date, default: Date.now
    },
    category : {
        type: Schema.Types.ObjectId, ref: 'Category' 
    },
    client : {
        type: Schema.Types.ObjectId, ref: 'Client' 
    },
    comments : [{
        type: Schema.Types.ObjectId, ref: 'Comment' 
    }]
  });

exports.PostSchema = PostSchema;
module.exports = mongoose.model('Post', PostSchema);

